### PR TITLE
update dynamic charts and images

### PR DESCRIPTION
**Pull request by sorenoid**
Related to issue: https://devtopia.esri.com/runtime/kotlin/issues/4215

### Description:

Update media charts when DynamicEntity updates
Update media images according to refresh interval

### Summary of changes:

- refactor creation of PopupMediaState
- wire in the newly instantiated PopupMedia to the PopupMediaState object when dynamic entity updates. After expressions are evaluated all the instances in the `Popup.evaluatedElements` list are newly instantiated.
- animate updates to charts, images, and FieldsPopupElements.

### Pre-merge Checklist

- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/) Job for this PR has been run
  - [x] link: https://runtime-kotlin.esri.com/view/vTest/job/vtest/job/toolkit/39/
- Unit and/or integration tests have been added to exercise this PR's logic, and the tests are passing:
  - [ ] Yes
  - [ ] No

### CODE REVIEW

File: toolkit/popup/src/main/java/com/arcgismaps/toolkit/popup/internal/element/media/MediaElementState.kt [hunk 1]
```diff
@@ -67,53 +72,30 @@ internal class MediaElementState(
...
     )
 
+    /**
+     * Update the PopupMedia so that a new chart image can be acquired. Only necessary
+     * for DynamicEntity Popups.
+     *
+     * @param newElement the new MediaPopupElement which contains the new PopupMedia
+     * @param scope the current CoroutineScope of the Composition.
+     */
+    fun updateMediaElement(newElement: MediaPopupElement, scope: CoroutineScope) {
```
Comment (eri9000) [inline toolkit/popup/src/main/java/com/arcgismaps/toolkit/popup/internal/element/media/MediaElementState.kt:91]:
```suggestion
    internal fun updateMediaElement(newElement: MediaPopupElement, scope: CoroutineScope) {
```

File: toolkit/popup/src/main/java/com/arcgismaps/toolkit/popup/internal/element/media/MediaElementState.kt [hunk 2]
```diff
@@ -183,9 +165,17 @@ internal fun rememberMediaElementState(
...
+        val geoElement = popup.geoElement
+        if (geoElement is DynamicEntity) {
+            // For dynamic entities
+            // update chart providers to use the new instances of PopupMedia to reacquire updated charts.
+            updateMediaElement(element, scope)
+        }
     }
 }
 
+
```
Comment (eri9000) [inline toolkit/popup/src/main/java/com/arcgismaps/toolkit/popup/internal/element/media/MediaElementState.kt:178]: extra line

File: toolkit/popup/src/main/java/com/arcgismaps/toolkit/popup/internal/element/media/MediaTile.kt [hunk 3]
```diff
@@ -130,50 +136,55 @@ internal fun MediaView(
...
+                contentDescription = title,
+                contentScale = ContentScale.FillBounds,
+                modifier = modifier.fillMaxSize(),
+                alignment = Alignment.Center,
+                alpha = DefaultAlpha,
+                colorFilter = null
             )
-            Caption(
-                text = caption,
+            if (title == "speed with time")
```
Comment (eri9000) [inline toolkit/popup/src/main/java/com/arcgismaps/toolkit/popup/internal/element/media/MediaTile.kt:151]: is this needed?

**Review by eri9000 — COMMENTED** [review]
LGTM 👍🏼 
a few small suggestions

File: toolkit/popup/src/main/java/com/arcgismaps/toolkit/popup/internal/element/media/MediaElementState.kt (hunk 1 above)
Comment (sorenoid) [reply toolkit/popup/src/main/java/com/arcgismaps/toolkit/popup/internal/element/media/MediaElementState.kt:91]: the class is internal.

**Review by eri9000 — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**